app/models/grda_warehouse/tasks/README_FOR_IDENTIFY_DUPLICATES.md

```diff
@@ -0,0 +1,149 @@
+```mermaid
+graph TD
+    subgraph DS1["HMIS Data Source 1"]
+        B["Source Client"]
+        B --> E["WarehouseClient"]
+    end
+
+    subgraph DS2["HMIS Data Source 2"]
+        D["Source Client"]
+        D --> F["WarehouseClient"]
+    end
+
+
+    subgraph DSW["Warehouse Data Source"]
+        H["Destination Client<br/>(data_source_id: 99)"]
+    end
+
+    E --> H
+    F --> H
+```
+
+## Two Main Operations
+
+The system provides two distinct operations that handle different deduplication scenarios:
+
+### Operation 1: `identify_duplicates` - Process New/Unprocessed Clients
+
+**Purpose**: Handles newly imported clients that need to be linked to existing destinations or have new destinations created
+
+**When it runs**:
+- Automatically when new clients are created
+- During data imports
+- Via manual execution
+
+**What it does**:
+1. Restores previously deleted destinations that still have active source clients
+2. Finds source clients that haven't been processed yet
+3. Matches them against existing destinations using matching criteria
+4. Either links to existing destinations or creates new ones
+5. Updates destination records with enriched data from sources
+
+### Operation 2: `match_existing!` - Merge Existing Destinations
+
+**Purpose**: Re-evaluates existing destination clients when their identifying information changes
+
+**When it runs**:
+- When client personally identifiable information is updated
+- Via manual execution for reconciliation
+
+**What it does**:
+1. Identifies destination clients that should now be considered the same person
+2. Handles complex merge scenarios and chains
+3. Respects system limits on destination client size
+4. Performs merges with proper cleanup
+5. Rebuilds affected service history records
+
+### Why Two Separate Operations?
+
+**Logical Separation**:
+- **New client processing**: Handles incremental addition of clients from data sources
+- **Existing client reconciliation**: Handles changes to identifying information that affect existing relationships
+
+**Performance Benefits**:
+- Avoids expensive re-evaluation of all relationships when processing new clients
+- Separates costly merge operations from routine client processing
+- Enables efficient batch processing of imports
+- Prevents performance degradation as the system scales
+
+## Matching Criteria
+
+The system requires **2 of 3** exact matches across these normalized fields:
+
+### Social Security Number
+- Must pass validation checks
+- Excludes obvious test values and placeholders
+
+### Full Name
+- Normalized to handle variations in formatting and accented characters
+- Strips non-alphanumeric characters
+
+### Date of Birth
+- Must be reasonable (after 1920)
+- Requires exact date match
+
+## Configuration and Constraints
+
+### System Controls
+- Master switches to enable/disable automatic processing
+- Thresholds for automatic acceptance/rejection of matches
+- Limits on maximum sources per destination client
```

Comment:
```suggestion
- Configuration to enable/disable automatic processing
```

- The thresholds aren't related to identify duplicates but to the probabilistic matching algorithm.
- Limits are not currently configurable (so not sure we want to list them here.)